app/webpacker/components/Posts/PostForm.jsx

```diff
@@ -0,0 +1,141 @@
+import React, { useCallback, useState } from 'react';
+import {
+  Button, Form, FormField, Header, Message,
+} from 'semantic-ui-react';
+import { useMutation } from '@tanstack/react-query';
+import I18n from '../../lib/i18n';
+import useInputState from '../../lib/hooks/useInputState';
+import useCheckboxState from '../../lib/hooks/useCheckboxState';
+import MarkdownEditor from '../wca/FormBuilder/input/MarkdownEditor';
+import { createPost, deletePost, editPost } from './api/posts';
+import { useConfirm } from '../../lib/providers/ConfirmProvider';
+import UtcDatePicker from '../wca/UtcDatePicker';
+import I18nHTMLTranslate from '../I18nHTMLTranslate';
+
+export default function PostForm({
+  header, allTags, post,
+}) {
+  const [formPost, setFormPost] = useState(post);
+  const [formTitle, setFormTitle] = useInputState(formPost?.title ?? '');
+  const [formBody, setFormBody] = useInputState(formPost?.body ?? '');
+  const [formTags, setFormTags] = useState(formPost?.tags_array ?? []);
+  const [formIsStickied, setFormIsStickied] = useCheckboxState(formPost?.sticky ?? false);
+  const [formShowOnHomePage, setFormShowOnHomePage] = useCheckboxState(formPost?.show_on_homepage ?? true);
+  const [unstickAt, setUnstickAt] = useState(formPost?.unstick_at ?? null);
```

Comment:
Feels weird to have states for both the (form)post, and most of the properties of the post. Besides initializing the other states (which happens initially, so doesn't need `formPost` to be a state), `formPost` is only used to check whether we're creating or editing, get the id (if we're editing), and show the url at the end.

It seems like we could get rid of `formPost` and instead just have 2 more states for the post id and url; then update those 2 states in the `onSuccess` functions below.

Reply:
yeah makes sense